ic-http/src/mock.rs

```diff
@@ -8,7 +8,8 @@ use std::time::Duration;
 #[derive(Clone)]
 pub(crate) struct Mock {
     pub(crate) request: CanisterHttpRequestArgument,
-    response: HttpResponse,
+    response: Option<HttpResponse>,
+    error: Option<(RejectionCode, String)>,
```

Comment:
How about we have a single field `result` which would be `Option<Result<HttpResponse, (RejectionCode, String)>>`. The reason is that this way it's more clear that it can be either one or another and it's enforced through the type system.

Reply:
Done in #185, good idea, thank you!